src/st/e_collect.h

```diff
@@ -100,14 +100,20 @@ u8* g_SubweaponAnimPrizeDrop[] = {
 };
 
 static s16 D_80180EB8[] = {-6, 4, 0, -8};
-#if defined VERSION_BETA || STAGE == STAGE_ST0
-// This is weird, the values have to go in later.
-// Note that this array is in rodata. Other overlays have it in data.
-static const s8 c_HeartPrizes[4][2];
-#else
+#if !defined(VERSION_BETA) && STAGE != STAGE_ST0
+// Note that this array is in data. MAD/ST0 have it in rodata.
 s8 c_HeartPrizes[] = {1, 5};
 #endif
 
+// from another file
+extern u16 g_EInitObtainable[];
+
+#include "prize_drop_fall.h"
+
+#include "prize_drop_fall2.h"
+
+#include "collect_heart.h"
```

Comment:
Had to shift some of the code around so it's interleved with data, since ST0/beta load their data inline so this needs to happen before `g_ExplosionYVelocities` etc. below.

I couldn't find any other way to accomplish this, any other way would put `heartPrizes` / `mad_unknown` at the end of the data imports